articles/precarious-data.md

```diff
@@ -1,51 +1,217 @@
+* If someone on my local network has a copy of the content, the link doesn't let me get the content from that alternate local source. It only lets me get the content from the location identified by the link. This is especially painful if my internet connection is slow or if I don't have an internet connection at all.
 
 ## The Solution: Identify Information by its Fingerprint, not its Location
 
-The alternative is to identify content by its "fingerprint" rather than identifying it by its location. That way, when someone says "Look at the content with this fingerprint" you can get it from anyone who has a copy of the content. To do this, we identify content by its cryptographic fingerprint, known as a cryptographic hash.
+The alternative is to identify content by its "fingerprint" rather than identifying it by its location. That way, when someone says "Look at the content with this fingerprint" you can get it from anyone who has a copy of the content. To do this, we identify content by its cryptographic hash. When we identify content in this way, using the content's fingerprint instead of its location to identify it, this is called _content-addressing_.
+The cryptographic hash for a piece of content never changes, which means **content addressing guarantees that the links will always return the same content**, regardless of _where_ I retrieve the content from.  You can do it by using a _content-addressed_ protocol like IPFS instead of using a _location-addressed_ protocol like HTTP.
+
+This decentralized approach radically increases the durability of data. It ensures that data will not become endangered as long as anyone is still relying on it because anyone can hold a valid copy of the data they care about. If you hold a copy of a dataset on any of your devices you become part of the network of stewards who protect if from being lost. You won't have to worry about whether someone is going to turn off the servers where your data are hosted because _you are one of the hosts_. You and your peers hold the data among yourselves and are able to share the data directly with each other without relying on centralized points of failure.
 
-The cryptographic fingerprint for a piece of content never changes, which means **content addressing guarantees that the links will always return the same content**, regardless of _where_ I retrieve the content from.
+Decentralization also increases the integrity of data because links are content-addressed. This means we can validate data by checking the data's fingerprints against the links. That kind of validation is impossible with location-addressed links. This is especially powerful on the large scale, where thousands of datasets reference each other millions of times. With location-addressed links, all of those connections are brittle. With content-addressed links, the connections become resilient.
 
```

Comment:
Note that you need to have a protocol which ensures that a hash always points to the data it is derived from. The decentralized web is basically just a bunch of people who ensure that they are sharing data using these protocols. 

I don't think this is a given, and people may ask. Might be a side point, however, but I don't think it is enough to just say 'hashes point to the content' - you need to know that the system you suggest ensures that this also always works.